arch/xtensa/core/xtensa-asm2.c

```diff
@@ -27,23 +28,23 @@ void *xtensa_init_stack(int *stack_top,
 
 	memset(bsa, 0, bsasz);
 
-	bsa[BSA_PC_OFF/4] = entry;
+	bsa[BSA_PC_OFF/4] = _thread_entry;
```

Comment:
Oh jeez. I missed this in the initial implementation.
I'm a little amazed our test cases pass, one of the essential things _thread_entry does is call k_thread_abort() after main() returns.
Good stuff.